sklearn/tests/test_base.py

```diff
@@ -359,3 +357,81 @@ def test_pickle_version_warning():
     # check that no warning is raised for external estimators
     TreeNoVersion.__module__ = "notsklearn"
     assert_no_warnings(pickle.loads, tree_pickle_noversion)
+
+
+class DontPickleCacheMixin(object):
+    def __getstate__(self):
+        data = self.__dict__.copy()
+        data["_cache"] = None
+        return data
+
+
+class MultiInheritanceEstimator(BaseEstimator, DontPickleCacheMixin):
+    def __init__(self, b=5):
+        self.b = b
+        self._cache = None
+
+    @property
+    def cache(self):
+        if not self._cache:
+            self._cache = "some value"
+        return self._cache
+
+
+class SingleInheritanceEstimator(BaseEstimator):
+    def __init__(self, b=5):
+        self.b = b
+        self._cache = None
+
+    def __getstate__(self):
+        data = self.__dict__.copy()
+        data["_cache"] = None
+        return data
+
+    @property
+    def cache(self):
+        if not self._cache:
+            self._cache = "some value"
+        return self._cache
+
+
+class TestPicklingConstraints(object):
```

Comment:
1. Addressed in 0a5541435b343454d79e44a7c3149908d82c43e7
2. since the tests are about estimators, isn't it (implicitly) clear, that `BaseEstimator` is involved. Also, the module being `sklearn.base`.